### PR TITLE
Add items under "mesh" menu

### LINKED ISSUE #324: Improve viewer menu
The viewer menu should be redesigned.  In contrast to the current 5 menus

image

there should be 6 menus in the menu bar:

image

The hierarchical list of the menu items:

* File
  * Load mesh file
* View (the original Camera)
  * Orbit camera
  * First-person camera
* One (the original Euler)
  * Load Euler
* Mesh
  * Load NACA airfoil
  * Sample meshes
* Add-on (the original App)
  * (Empty for now)
* Windows
  * (Empty for now)

A new add-on/plug-in system will be implemented and the add-ons/plug-ins should be placed under the Add-on menu.

### PR DESCRIPTION
**Pull request by yungyuc**
For issue #324

Update the following items under "mesh" menu:

* Sample: 3D mesh of mixed elements
* Sample: NACA 4-digit

Add the following items under the menu.  They are migrated from the module "modmesh.app.sample_mesh":

* Sample: mesh of a triangle (2D)
* Sample: mesh of a tetrahedron (3D)
* Sample: mesh of "solvcon" text in 2D

image

The new code is in the Python modules `modmesh.gui.sample_mesh` and `modmesh.gui.naca`.

A new C++ Qt class `modmesh::RPythonAction` is added for the menu items.

### CODE REVIEW

File: cpp/modmesh/view/RAction.hpp [hunk 1]
```diff
@@ -47,6 +47,22 @@ class RAction
         QObject * parent = nullptr);
 }; /* end class RAction */
 
+class RPythonAction
```
Comment (yungyuc) [inline cpp/modmesh/view/RAction.hpp:50]: A new C++ Qt class to call a Python function.

File: cpp/modmesh/view/RManager.cpp [hunk 2]
```diff
@@ -212,19 +212,31 @@ void RManager::setUpMenu()
 
     m_oneMenu = m_mainWindow->menuBar()->addMenu(QString("One"));
     m_oneMenu->addAction(new RAppAction(
-        QString("Load euler1d"),
-        QString("Load euler1d"),
+        QString("Euler solver"),
```
Comment (yungyuc) [inline cpp/modmesh/view/RManager.cpp:215]: Piggyback on this PR to rename the Euler solver menu item.  It (should) still use the "app".

File: cpp/modmesh/view/RManager.cpp [hunk 3]
```diff
@@ -212,19 +212,31 @@ void RManager::setUpMenu()
...
     m_meshMenu = m_mainWindow->menuBar()->addMenu(QString("Mesh"));
-    m_meshMenu->addAction(new RAppAction(
-        QString("Load mh_3dmix"),
-        QString("Load mh_3dmix"),
-        QString("modmesh.app.mh_3dmix")));
-    m_meshMenu->addAction(new RAppAction(
-        QString("Load naca"),
-        QString("Load naca"),
-        QString("modmesh.app.naca")));
+    m_meshMenu->addAction(new RPythonAction(
```
Comment (yungyuc) [inline cpp/modmesh/view/RManager.cpp:220]: Use the new `RPythonAction` for the mesh menu items.

File: modmesh/gui/__init__.py
Comment (yungyuc) [inline modmesh/gui/__init__.py:1]: Remove the `mh_3dmix` app.  @YenPeiChen07 FYI.

File: modmesh/gui/naca.py [hunk 4]
```diff
@@ -108,37 +107,15 @@ def draw_naca4(world, number, npoint, fac, off_x, off_y, **kw):
...
     """
     A simple example for drawing a couple of cubic Bezier curves.
     """
     w = core.WorldFp64()
     draw_naca4(w, number='0012', npoint=101, fac=5.0, off_x=0.0, off_y=2.0,
                open_trailing_edge=False, cosine_spacing=True)
-    return w
-
-
-def load_app():
```
Comment (yungyuc) [inline modmesh/gui/naca.py:121]: Turn the example NACA 4-digit airfoil shape a simple function.  An "app" is an overkill.

**Review by yungyuc — COMMENTED** [review]
@j8xixo12 could you please help take a look?

File: modmesh/gui/sample_mesh.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        (4, 99, 100, 128, 127),
+        (4, 100, 101, 129, 128),
+        (4, 102, 103, 131, 130),
+        (4, 103, 104, 132, 131),
+        (4, 104, 105, 133, 132),
+        (4, 106, 107, 135, 134),
+        (4, 110, 111, 137, 136),
+        (4, 111, 112, 138, 137),
+        (4, 112, 113, 139, 138)
+    ]
```
Comment (ThreeMonth03) [inline modmesh/gui/sample_mesh.py:163]: How do you generate the testcase?
The example looks excessive long.

**Review by ThreeMonth03 — COMMENTED** [review]
The coding style and encapsulation are decent, 
but the example is not easy to read.

File: modmesh/gui/sample_mesh.py (hunk 5 above)
Comment (yungyuc) [reply modmesh/gui/sample_mesh.py:163]: I didn't.  Kudos to @Yu-Rou-Weng who did it by hands.

The code here is not exactly test cases.  They are examples.

**Comment by yungyuc** [comment]
> The coding style and encapsulation are decent, 
> 
> but the example is not easy to read.

No.  They are not easy to read.  They are supposed to be viewed in the viewer.

**Comment by yungyuc** [comment]
I am going to merge.  @j8xixo12 feel free to ask for modification.  I will address in a future PR.

**Pull request merged**